decompose/src/webMain/kotlin/com/arkivanov/decompose/router/stack/webhistory/DefaultWebHistoryController.kt

```diff
@@ -152,22 +154,44 @@ class DefaultWebHistoryController internal constructor(
 
         fun onPopState(state: String?) {
             val newData = state?.let(::deserializeItems) ?: return
-            val newConfiguration = deserializeConfiguration(json = newData.last().configurationJson)
+            val configurations = newData.map {
+                deserializeConfiguration(json = it.configurationJson)
+            }
+
+            doOnPopState(newData, configurations.last(), onWebNavigation(configurations))
+        }
 
+        private fun doOnPopState(newData: List<PageItem>, newConfiguration: C, navigationAllowed: Boolean) {
             isStateObserverEnabled = false
 
             navigator.navigate { stack ->
```

Comment:
Could you please elaborate why do we need to navigate at all if the retuned result from `onWebNavigation` (i.e. `navigationAllowed` flag) is `false`? And why do we need all these changes in the logic? I thought the only change needed is to just do nothing if the result is `false` and keep doing all the current stuff (i.e. navigate) if the result is `true`.